components/suggestion/search_channel_suggestion/search_channel_suggestion.tsx

```diff
@@ -3,22 +3,23 @@
 import React from 'react';
 
 import {getUserIdFromChannelName} from 'mattermost-redux/utils/channel_utils';
-
 import {imageURLForUser} from 'utils/utils.jsx';
 import Constants from 'utils/constants';
 import Avatar from 'components/widgets/users/avatar';
 import BotBadge from 'components/widgets/badges/bot_badge';
+import Suggestion from '../suggestion';
 
-import Suggestion from '../suggestion.jsx';
+import {UserProfile} from 'mattermost-redux/types/users';
+import {Channel} from 'mattermost-redux/types/channels';
 
-function itemToName(item, currentUser) {
-    let itemMarkup = (item);
+function itemToName(item: Channel, currentUser: UserProfile) {
+    let itemMarkup;
 
     if (item.type === Constants.DM_CHANNEL) {
         const profilePicture = (
             <Avatar
-                username={item.username}
-                url={imageURLForUser(getUserIdFromChannelName(currentUser, item.name), item.last_picture_update)}
+                username={currentUser.username}
```

Comment:
I'm not sure why we're changing the logic here, can you elaborate as to why?

Reply:
Hi @devinbinnie , since object  item has data type "Channel" , it doesn't have property username.
Username in tag "Avatar" should refer o name of user, right ? 

So I change the code from `username={item.username}` to be  `username={currentUser.username}`.

Reply:
Hmm, well that would create a logic change which is not what we want. I think `item` is actually of type `UserProfile` as well since that would make sense with the fields that are used.

Reply:
Did some further investigating, it looks like we only get `Channel` objects there and they don't have the `last_picture_update` field, so it will always be undefined. Maybe we just take out the fields that are missing and leave the rest of the logic the way it is?

Sorry for the confusion.

Reply:
Hi @devinbinnie ,so I will revert the **currentUser to be string again** and the logic of code will be like this ?

 ```
 function itemToName(item: Channel, currentUser: string) {
    let itemMarkup;


    if (item.type === Constants.DM_CHANNEL) {
        const profilePicture = (
            <Avatar
               // delete username={item.username} in <Avatar> since Channel.username is always undefined
                url={imageURLForUser(getUserIdFromChannelName(currentUser, item.name))} // will not set last_picture_update 
                size='sm'
            />
        );

        itemMarkup = (
            <React.Fragment>
                {profilePicture}
                <span className='ml-3'>{'@'}{item.display_name}</span>
            </React.Fragment>
        );
    }
    ....... other code ......
  }
````
  
and in file components/suggestion/search_channel_suggestion/index.ts, i will revert back 
`currentUser: getCurrentUser(state)` (UserProfile)  to be `currentUser: getCurrentUserId(state)` (string)  ?

Reply:
Yes I believe that's correct. Thanks for the hard work :)

Reply:
Hi @devinbinnie , I already change it. Thank you